Modules/services/Controller/ServicesprojectsController.php

```diff
@@ -677,6 +685,34 @@ public function getTaskFileAction($id_space, $id_task) {
         $this->render(['data' => $file]);
     }
 
+    public function openFileAction($id_space, $id_project, $id_task) {
+        $this->checkAuthorizationMenuSpace("services", $id_space, $_SESSION["id_user"]);
+        $taskModel = new SeTask();
+        $task = $taskModel->getById($id_space, $id_task);
+
+        // If private task, check if user is the owner of the task
+        if ($task['private'] == 1 && $task['id_owner'] != $_SESSION["id_user"]) {
```

Comment:
If user yes, else any manager should ve allowed

Reply:
You mean we're ok with this behaviour ?

Reply:
No

If private and role is user, then you need to check owner (what you do) but if role is >= manager, then you should have access

Reply:
we already have the `$this->checkAuthorizationMenuSpace("services", $id_space, $_SESSION["id_user"]);` above

Reply:
If private and we are manager, id owner may be different from session id user.

A manager creates the task/file, an other manager wants to access it

Reply:
I can modify the condition to "if role >= manager OR if session user is the task owner"

Reply:
We won t allow users (members of project of course) to create task.
Only to view them, if public, and later possibly comment on task or change their status.

Only managers can create tasks

Users must not be allowed to create/edit tasks for the moment

Reply:
I wanted to deal with authorising users to view projects in a future PR.
Currently, only roles authorised to access to services (generally managers and admins) can access to projects (read and write, no difference right now).